### PR TITLE
feat: check blob data availability

### LINKED ISSUE #765: Check blob data availability in fork choice `on_block` handler
See:
- https://eips.ethereum.org/EIPS/eip-4844
- https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/fork-choice.md#on_block

Depends on:
- https://github.com/lambdaclass/lambda_ethereum_consensus/issues/746
- Probably https://github.com/lambdaclass/lambda_ethereum_consensus/issues/754

Acceptance criteria:
- Fork choice spec tests should work for Deneb fork that use blobs: https://github.com/ethereum/consensus-specs/blob/dev/tests/formats/fork_choice/README.md#on_block-execution-step

### PR DESCRIPTION
**Pull request by MegaRedHand**
Closes #765

### CODE REVIEW

File: test/spec/runners/fork_choice.ex [hunk 1]
```diff
@@ -16,21 +17,9 @@ defmodule ForkChoiceTestRunner do
 
   use HardForkAliasInjection
 
-  # TODO: implement blob checks
```
Comment (mpaulucci) [inline test/spec/runners/fork_choice.ex:19]: 🎉

File: test/spec/tasks/generate_spec_tests.ex [hunk 2]
```diff
@@ -92,6 +92,7 @@ defmodule Mix.Tasks.GenerateSpecTests do
 
   defp generate_case(runner_module, testcase) do
     """
+      @tag :tmp_dir
```
Comment (mpaulucci) [inline test/spec/tasks/generate_spec_tests.ex:95]: whats this for?

Comment (MegaRedHand) [reply test/spec/tasks/generate_spec_tests.ex:95]: The [`tmp_dir`](https://hexdocs.pm/ex_unit/main/ExUnit.Case.html#module-tmp-dir) tag tells `ExUnit` it should create a temporary directory for the test and adds its path to the context with `tmp_dir` key. The blob tests reuse the same blocks on different testcases, which causes the tests to fail if some of those testcases aren't supposed to have blobs. That's why we need to nuke the DB after finishing each test.

**Review by mpaulucci — APPROVED** [review]

File: test/spec/tasks/generate_spec_tests.ex (hunk 2 above)
Comment (mpaulucci) [reply test/spec/tasks/generate_spec_tests.ex:95]: nice!

**Pull request merged**